lib/rules/strict.js

```diff
@@ -82,46 +82,46 @@ module.exports = function(context) {
     }
 
     //--------------------------------------------------------------------------
-    // "never" mode
+    // If this is modules, all "use strict" directives are unnecessary.
     //--------------------------------------------------------------------------
 
-    if (mode === "never") {
+    if (context.parserOptions.sourceType === "module") {
```

Comment:
You should check the program node for `sourceType`, as not all parsers will support parsing as a module using parser options.